go/vt/tabletserver/query_executor.go

```diff
@@ -198,15 +198,15 @@ func (qre *QueryExecutor) execAsTransaction(f func(conn *TxConnection) (*sqltype
 	reply, err = f(conn)
 
 	if err != nil {
+		defer qre.logStats.AddRewrittenSQL("rollback", time.Now())
```

Comment:
I think it's better to make this verbose. Problem with defer is that it will end up including time taken by any other operation we add later to this function.
```
start := time.Now()
operation()
qre.logstats.AddRewritten(..., start)
```
Same comment below.